python/wawenets/wawenets/stl_wrapper.py

```diff
@@ -0,0 +1,1117 @@
+        bool
+            whether or not the calculated file size ratio matches the target file size
+            ratio, and by proxy indicating if resampling was successful
+        """
+        # `filter` puts all of its output in `stderr`, hmm
+        stdout, stderr = self._call_command(command)
+
+        # because we can't naively check `stderr`, let's naively calculate size ratio
+        return self._check_filesize_ratio(in_path, out_path, target_ratio)
+
+    def down_48k_to_16k(self, in_path: Path, out_path: Path) -> bool:
+        """
+        resamples the input raw (pcm) file, with sample rate assumed to be 48,000
+        samp / sec to 16,000 samp / sec and writes the result.
+
+        Parameters
+        ----------
+        in_path : Path
+            file path to the raw (pcm) file with a sample rate of 48,000 samp / sec
+        out_path : Path
+            file path where the resampled raw (pcm) file should be written
+
+        Returns
+        -------
+        bool
+            whether the resampling operation was successful
+        """
+        target_ratio = 3.0
+        command = [
+            self.filter_path,
+            "-q",
+            "-down",
+            "HQ3",
+            str(in_path),
+            str(out_path),
+        ]
+
+        return self._call_filter(command, target_ratio, in_path, out_path)
+
+    def down_32k_to_16k(self, in_path: Path, out_path: Path) -> bool:
+        """
+        resamples the input raw (pcm) file, with sample rate assumed to be 32,000
+        samp / sec to 16,000 samp / sec and writes the result.
+
+        Parameters
+        ----------
+        in_path : Path
+            file path to the raw (pcm) file with a sample rate of 32,000 samp / sec
+        out_path : Path
+            file path where the resampled raw (pcm) file should be written
+
+        Returns
+        -------
+        bool
+            whether the resampling operation was successful
+        """
+        target_ratio = 2.0
+        command = [
+            self.filter_path,
+            "-q",
+            "-down",
+            "HQ2",
+            str(in_path),
+            str(out_path),
+        ]
+
+        return self._call_filter(command, target_ratio, in_path, out_path)
+
+    def down_24k_to_16k(self, in_path: Path, out_path: Path) -> bool:
+        """
+        resamples the input raw (pcm) file, with sample rate assumed to be 24,000
+        samp / sec to 16,000 samp / sec and writes the result.
+
+        Parameters
+        ----------
+        in_path : Path
+            file path to the raw (pcm) file with a sample rate of 24,000 samp / sec
+        out_path : Path
+            file path where the resampled raw (pcm) file should be written
+
+        Returns
+        -------
+        bool
+            whether the resampling operation was successful
+        """
+        with tempfile.NamedTemporaryFile() as temporary:
+            temp_path = Path(temporary.name)
+            target_ratio = 0.5
```

Comment:
Should this be `1.5`?

Reply:
no, this is correct as-is. the way this works is that it upsamples the 24k samp/sec file to 48k samp/sec before calling `down_48k_to_16k` (which has its own target ratio). so for this intermediary step of upsampling by two, the target ratio should be 0.5.